File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/productlist/ProductListContract.java
package com.gerigorila.agenticmigrationdemo.presentation.productlist;

import com.gerigorila.agenticmigrationdemo.domain.model.Product;

import java.util.List;

public interface ProductListContract {

    interface View {
        void showProducts(List<Product> products);
        void navigateToProductDetail(int productId);
    }

    interface Presenter {
        void loadProducts();
        void onProductClicked(int productId);
    }
}

File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/productlist/ProductListContract.kt
package com.gerigorila.agenticmigrationdemo.presentation.productlist

import com.gerigorila.agenticmigrationdemo.domain.model.Product

interface ProductListContract {

    interface View {
        fun showProducts(products: List<Product>)
        fun navigateToProductDetail(productId: Int)
    }

    interface Presenter {
        fun loadProducts()
        fun onProductClicked(productId: Int)
    }
}
